src/eth/TransactionManager.js

```diff
@@ -57,7 +57,20 @@ export default class TransactionManager extends PublicService {
       (async () => {
         // so we do our async operations inside this immediately-executed
         // async function.
-        const txOptions = await this._buildTransactionOptions(options);
+        let data = contract.interface.functions[method](...args).data;
```

Comment:
it would be cleaner to change this to 
```js
const txOptions = await this._buildTransactionOptions(options, contract);
```
and then have all the rest of the changes inside `_buildTransactionOptions`